todo_app/app.py

```diff
@@ -62,20 +74,38 @@ def get_cards():
         all_cards = [ToDoCard.from_mongo_card(card) for card in cards.find()]
         view_model = ViewModel(all_cards)
         return render_template('cards.html', view_model= view_model)
+
+    @app.route('/auth_error')
+    def auth_error_page():
+        return render_template('auth_code_error.html')
+    
+    @app.route('/user_response_error')
+    def user_response_error():
+        return render_template('user_response_error.html')
+
     
     @app.route('/login/callback')
     def verification():
-        auth_code = request.args['code']
+        try:
+            auth_code = request.args['code']
+        except KeyError:
+            app.logger.error("auth_code missing")
+            return redirect(url_for('auth_error_page'))
```

Comment:
I know we discussed on Slack, but this is a nice approach to handling the error - we log the error, and spell out the issue to the user